apps/src/templates/GDPRDialog.jsx

```diff
@@ -0,0 +1,82 @@
+import React, {Component, PropTypes} from 'react';
+import Button from './Button';
+import i18n from "@cdo/locale";
+import BaseDialog from './BaseDialog';
+import DialogFooter from "./teacherDashboard/DialogFooter";
+import {pegasus} from '@cdo/apps/lib/util/urlHelpers';
+
+const styles = {
+  dialog: {
+    paddingLeft: 20,
+    paddingRight: 20,
+    paddingBottom: 20
+  },
+  instructions: {
+    marginTop: 20
+  }
+};
+
+export default class GDPRDialog extends Component {
+  static propTypes = {
+    isDialogOpen: PropTypes.bool.isRequired,
+    currentUserId: PropTypes.number.isRequired,
+    requestIp: PropTypes.string.isRequired,
+    // If we're coming from pegasus, studioUrlPrefix is passed in as a prop and used to construct the logout url.
+    studioUrlPrefix: PropTypes.string
+  };
+
+  state = {
+    isDialogOpen: this.props.isDialogOpen,
+  };
+
+  handleYesClick = () => {
+    this.setState({isDialogOpen: false});
+    $.post(
+      `/api/v1/users/${this.props.currentUserId}/accept_data_transfer_agreement`, {
```

Comment:
Did you decide on this API approach rather than a PATCH?

Reply:
I think so.  I'm interested in pros/cons either way.  This approach is consistent with dismissing and tracking the `UITips` on the teacher homepage and courses, which felt conceptually similar to dismissing the dialog here and setting user properties. It also didn't make as much sense to me for this to be something `registrations_controller` handles.